client.go

```diff
@@ -372,6 +372,52 @@ func (cli *Client) Versions() (resp *RespVersions, err error) {
 	return
 }
 
+// PublicRooms returns the list of public rooms on target server. Does not require Auth. See http://matrix.org/speculator/spec/HEAD/client_server/unstable.html#get-matrix-client-unstable-publicrooms
+func (cli *Client) PublicRooms(limit int, since string, server string) (resp *RespPublicRooms, err error) {
+	args := map[string]string{}
+
+	if limit != 0 {
+		args["limit"] = strconv.Itoa(limit)
+	}
+	if since != "" {
+		args["since"] = since
+	}
+	if server != "" {
+		args["server"] = server
+	}
+
+	urlPath := cli.BuildURLWithQuery([]string{"publicRooms"}, args)
+	err = cli.MakeRequest("GET", urlPath, nil, &resp)
+	return
+}
+
+// PublicRoomsFiltered returns a subset of PublicRooms filtered server side. See http://matrix.org/speculator/spec/HEAD/client_server/unstable.html#post-matrix-client-unstable-publicrooms
```

Comment:
```suggestion
// PublicRoomsFiltered returns a subset of PublicRooms filtered server side.
// See https://matrix.org/docs/spec/client_server/r0.6.0#post-matrix-client-r0-publicrooms
```

Also going to run into the auth problem.